### PR TITLE
Issue #1868 and Issue #1984

### PR DESCRIPTION
**Pull request by beeduck**
Addressing issue #1868, I've added the sound effect for thunder. It is created with a fixed volume to be realistically heard over the sound of the rain.

I noticed that lightning is currently created at a fixed location (0, 0, 0), I was wondering if anyone is currently working on the TODO of changing lightning to be cast randomly near player locations. If not, I'd like to try and fix that as well. This would also probably change the fixed volume of the thunder sound effect.

I've also addressed issue #1984. Door open/close and sound effect handling was only looking for basic wooden door types.

### CODE REVIEW

**Comment by bearbin** [comment]
Looks great!

If you could add yourself to the CONTRIBUTORS file, then I think this could be merged.

https://github.com/mc-server/MCServer/blob/master/CONTRIBUTORS

File: src/Blocks/BlockDoor.cpp [hunk 1]
```diff
@@ -50,10 +50,20 @@ void cBlockDoorHandler::OnUse(cChunkInterface & a_ChunkInterface, cWorldInterfac
 	UNUSED(a_CursorY);
 	UNUSED(a_CursorZ);
 
-	if (a_ChunkInterface.GetBlock(a_BlockX, a_BlockY, a_BlockZ) == E_BLOCK_WOODEN_DOOR)
+	switch (a_ChunkInterface.GetBlock(a_BlockX, a_BlockY, a_BlockZ))
```
Comment (worktycho) [inline src/Blocks/BlockDoor.cpp]: This switch should have a default case, to cover future new door blocks.

File: src/World.cpp [hunk 2]
```diff
@@ -2452,6 +2452,7 @@ void cWorld::BroadcastTeleportEntity(const cEntity & a_Entity, const cClientHand
 void cWorld::BroadcastThunderbolt(int a_BlockX, int a_BlockY, int a_BlockZ, const cClientHandle * a_Exclude)
 {
 	m_ChunkMap->BroadcastThunderbolt(a_BlockX, a_BlockY, a_BlockZ, a_Exclude);
+	BroadcastSoundEffect("ambient.weather.thunder", a_BlockX, a_BlockY, a_BlockZ, 50, 1);
```
Comment (worktycho) [inline src/World.cpp]: This should probably be in CastThunderbolt, not broadcast thunderbolt because the broadcasts are tied to the packets.

Comment (NiLSPACE) [inline src/World.cpp]: I disagree. Some plugins might want it to just make a bright flash in the sky (for example for some kind of minigame). Not to make the sound with it.

Comment (NiLSPACE) [inline src/World.cpp]: I think there should be a way to make the lightning flash without the sound. It could be used in for example minigames.

Comment (worktycho) [inline src/World.cpp]: Which is why it should be in CastThunderbolt. CastThunderbolt is for is a plugin wants to simulate a minecraft thunderbolt with everything that comes with that. BroadcastThunderbolt should just be to send the flash.

Comment (NiLSPACE) [inline src/World.cpp]: Yeah, I didn't make it clear that I agreed with you. We do have to export the BroadcastThunderbolt function as well though.

Comment (worktycho) [inline src/World.cpp]: I'm happy exporting it. But not until after I've moved it to cBroadcaster.

File: src/Blocks/BlockDoor.cpp (hunk 1 above)
Comment (madmaxoft) [inline src/Blocks/BlockDoor.cpp]: With an `ASSERT(!"Unhandled door block type");` in it so that it is quickly found when such a case happens.

**Comment by beeduck** [comment]
Okay I'll make the default door case and thunder sound effect change a little later today

**Comment by beeduck** [comment]
Are there any other changes you need me to make to this before it can be merged?

File: CONTRIBUTORS [hunk 3]
```diff
@@ -35,5 +35,6 @@ WebFreak001
 worktycho
 xoft
 Yeeeeezus (Donated AlchemistVillage prefabs)
+beeduck
```
Comment (bearbin) [inline CONTRIBUTORS]: You should be near the front, this list is alpha-sorted.

**Comment by bearbin** [comment]
I don't think you've addressed @worktycho and @xoft's comments yet?

**Comment by worktycho** [comment]
@bearbin, there addressed, The only issue is the style of the default. I like it, but I suspect @madmaxoft is going to want braces. So I'm waiting on his comments.

File: src/Blocks/BlockDoor.cpp [hunk 4]
```diff
@@ -50,10 +50,22 @@ void cBlockDoorHandler::OnUse(cChunkInterface & a_ChunkInterface, cWorldInterfac
...
 	{
-		ChangeDoor(a_ChunkInterface, a_BlockX, a_BlockY, a_BlockZ);
-		a_Player->GetWorld()->BroadcastSoundParticleEffect(1003, a_BlockX, a_BlockY, a_BlockZ, 0, a_Player->GetClientHandle());
+		default:
+			ASSERT(!"Unhandled door block type");
```
Comment (madmaxoft) [inline src/Blocks/BlockDoor.cpp]: Braces: yes please :)

**Comment by madmaxoft** [comment]
Fix the braces and squash the unneeded commits, and we can merge.

**Comment by beeduck** [comment]
Alright, all the commits should be in order now

**Pull request merged**